src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializer.java

```diff
@@ -415,7 +415,9 @@ protected Object _deserializeUsingPropertyBased(final JsonParser p, final Deseri
         throws IOException
     {
         final PropertyBasedCreator creator = _propertyBasedCreator;
-        PropertyValueBuffer buffer = creator.startBuilding(p, ctxt, _objectIdReader);
+        PropertyValueBuffer buffer = (_anySetter != null)
+            ? creator.startBuildingWithAnySetter(p, ctxt, _objectIdReader, _anySetter)
+            : creator.startBuilding(p, ctxt, _objectIdReader);
```

Comment:
Ok, now I remember a bigger improvement that'd be great here; and could maybe remove need for that "is consumed" marker (I'll see how that code has changed, I think you had comments): instead of buffering "any properties", we should ideally collect them right along with regular ones and avoid buffering.
This would have multiple benefits -- not being repeated, avoiding actual (temporary) storage of buffered.
And maybe more importantly: for data format(s) like XML, where buffering has issues (`TokenBuffer` loses some information on things like if something is XML Attribute or element), it would just work better.

I know existing code did use buffering but maybe this could now be improved?

Reply:
After going through code, maybe this is not a big deal -- esp. XML case should be fine as we do not buffer stream (tokens) but only Java values. Would be nice to avoid, and perhaps we can achieve that eventually, but need not be done right now.